packages/react-devtools-inline/__tests__/__e2e__/inspecting-props.test.js

```diff
@@ -4,49 +4,131 @@ const {test, expect} = require('@playwright/test');
 const config = require('../../playwright.config');
 test.use(config);
 
-test.describe('Testing Todo-List App', () => {
-  let page, frameElementHandle, frame;
-  test.beforeAll(async ({browser}) => {
+test.describe('ListApp', () => {
+  let page;
+
+  test.beforeEach(async ({browser}) => {
     page = await browser.newPage();
+
     await page.goto('http://localhost:8080/e2e.html', {
       waitUntil: 'domcontentloaded',
     });
-    await page.waitForSelector('iframe#iframe');
-    frameElementHandle = await page.$('#iframe');
-    frame = await frameElementHandle.contentFrame();
+
+    await page.waitForSelector('#iframe');
   });
 
-  test('The Todo List should contain 3 items by default', async () => {
-    const list = frame.locator('.listitem');
-    await expect(list).toHaveCount(3);
+  test('The List should contain 3 items by default', async () => {
+    const appRowCount = await page.evaluate(async () => {
+      const {createTestNameSelector, findAllNodes} = window.REACT_DOM_APP;
+      const container = document.getElementById('iframe').contentDocument;
+      const rows = findAllNodes(container, [
+        createTestNameSelector('ListItem'),
+      ]);
+      return rows.length;
+    });
+    expect(appRowCount).toBe(3);
+
+    const devToolsRowCount = await page.evaluate(async () => {
+      const {
+        createTestNameSelector,
+        createTextSelector,
+        findAllNodes,
+      } = window.REACT_DOM_DEVTOOLS;
+      const container = document.getElementById('devtools');
+      const rows = findAllNodes(container, [
+        createTestNameSelector('ComponentTreeListItem'),
+        createTextSelector('ListItem'),
+      ]);
+      return rows.length;
+    });
+    expect(devToolsRowCount).toBe(3);
   });
 
-  test('Add another item Fourth to list', async () => {
-    await frame.type('.input', 'Fourth');
-    await frame.click('button.iconbutton');
-    const listItems = await frame.locator('.label');
-    await expect(listItems).toHaveText(['First', 'Second', 'Third', 'Fourth']);
+  test('New list items should be added to DevTools', async () => {
+    await page.evaluate(async () => {
+      const {createTestNameSelector, findAllNodes} = window.REACT_DOM_APP;
+      const container = document.getElementById('iframe').contentDocument;
+      const button = findAllNodes(container, [
+        createTestNameSelector('AddItemButton'),
```

Comment:
Interesting you can create your own selectors no need to alter HTML or JS code for adding IDs and classes. Great !